File: src/connectivity/network/netstack3/core/tcp/src/testutil.rs
// Copyright 2025 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use core::ops::Range;

use netstack3_base::{SackBlock, SackBlocks, SeqNum};

/// Creates a [`SackBlocks`] from the sequence number ranges represented as
/// `u32`s.
pub(crate) fn sack_blocks(iter: impl IntoIterator<Item = Range<u32>>) -> SackBlocks {
    iter.into_iter()
        .map(|Range { start, end }| {
            SackBlock::try_new(SeqNum::new(start), SeqNum::new(end)).unwrap()
        })
        .collect()
}

File: scripts/cog/tests/test_util.py
# Copyright 2026 The Fuchsia Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Tests for util."""

import unittest

import util


class TestUtil(unittest.TestCase):
    """Tests for util."""

    def test_sanitize_filename(self) -> None:
        """Test that sanitize_filename replaces invalid characters."""
        self.assertEqual(util.sanitize_filename("hello world"), "hello-world")
        self.assertEqual(util.sanitize_filename("hello/world"), "hello-world")
        self.assertEqual(util.sanitize_filename("hello..world"), "hello..world")
        self.assertEqual(util.sanitize_filename("hello-world"), "hello-world")
        self.assertEqual(util.sanitize_filename("hello_world"), "hello_world")
        self.assertEqual(util.sanitize_filename("Hello World"), "hello-world")
        self.assertEqual(
            util.sanitize_filename("hello!@#$%^&*()world"), "hello-world"
        )


if __name__ == "__main__":
    unittest.main()
